### PR TITLE
Allow passing a function to extrema

### PR DESCRIPTION
**Pull request by ararslan**
Currently `minimum` and `maximum` can accept a function argument, but `extrema` cannot. This makes it consistent.

~~I've set the `!!! compat` to Julia 1.1, but if this can't make it into the release then I can set it to 1.2 or whatever.~~ Set to 1.2.

### CODE REVIEW

**Review by StefanKarpinski — APPROVED** [review]

**Comment by StefanKarpinski** [comment]
Marking for triage just to discuss whether we're allowing backports of new features past the 1.1 release branch date.

File: base/multidimensional.jl [hunk 1]
```diff
@@ -1500,40 +1500,54 @@ julia> extrema(A, dims = (1,2))
...
-_extrema_dims(A::AbstractArray, ::Colon) = _extrema_itr(A)
+"""
+    extrema(f, A::AbstractArray; dims) -> Array{Tuple}
+
+Compute the minimum and maximum of `f` applied to each element in the given dimensions
+of `A`.
+
+!!! compat "Julia 1.1"
+    This method requires Julia 1.1 or later.
+```
```
Comment (ararslan) [inline base/multidimensional.jl]: Whoops, stray open code block here

**Comment by StefanKarpinski** [comment]
Should be better now.

**Comment by JeffBezanson** [comment]
No, I don't think we should backport new features. IMO backporting after branch is only to fix release blockers.

**Comment by ararslan** [comment]
32-bit Windows failure is the Profile thing.

**Pull request merged**